### PR TITLE
Alanren/tool check

### PR DESCRIPTION
**Pull request by alanrenmsft**
after discussion, we decided to do the tools check in notebooks, the ui will only need to show the required tools

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
> 		return localize('resourceDeployment.PythonDescription', 'Required by notebook feature');

Side note - this description doesn't seem to really be an actual description of the tool like the others. Intended?

---
Refers to: extensions/resource-deployment/src/services/tools/pythonTool.ts:16 in 5239fed. [](commit_id = 5239fed75a067c4c84a24ff9124a6f6e7edaac27, deletion_comment = False)

File: extensions/resource-deployment/src/ui/resourceDeploymentDialog.ts [hunk 1]
```diff
@@ -165,27 +160,17 @@ export class ResourceDeploymentDialog {
 	}
 
 	private updateTools(): void {
-		this.toolsService.getToolStatus(this.getCurrentProvider().requiredTools).then(toolStatus => {
-			let tableData = toolStatus.map(tool => {
-				return [tool.name, tool.description, tool.version, this.getToolStatusText(tool.status)];
+		// do a 10 ms delay to workaround the issue of first time load
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/ui/resourceDeploymentDialog.ts]: >// do a 10 ms delay to workaround the issue of first time load [](start = 2, length = 62)

What issue?

File: extensions/resource-deployment/src/services/toolsService.ts [hunk 2]
```diff
@@ -3,50 +3,32 @@
...
 	getToolByName(toolName: string): ITool | undefined {
 		if (toolName) {
-			for (let i = 0; i < ToolsService.SupportedTools.length; i++) {
-				if (toolName === ToolsService.SupportedTools[i].name) {
-					return ToolsService.SupportedTools[i];
+			for (let i = 0; i < this.SupportedTools.length; i++) {
+				if (toolName === this.SupportedTools[i].name) {
+					return this.SupportedTools[i];
 				}
 			}
```
Comment (Charles-Gagnon) [inline extensions/resource-deployment/src/services/toolsService.ts]: Could also just do 

```
return this.SupportedTools.find(t => t.name === toolName)
```

**Comment by alanrenmsft** [comment]
> are the texts are not final here, Dinakar is working on them. > return localize('resourceDeployment.PythonDescription', 'Required by notebook feature');
> 
> Side note - this description doesn't seem to really be an actual description of the tool like the others. Intended?
> 
> Refers to: extensions/resource-deployment/src/services/tools/pythonTool.ts:16 in [5239fed](https://github.com/microsoft/azuredatastudio/commit/5239fed75a067c4c84a24ff9124a6f6e7edaac27). [](commit_id = [5239fed](https://github.com/microsoft/azuredatastudio/commit/5239fed75a067c4c84a24ff9124a6f6e7edaac27), deletion_comment = False)
sorry, for some reason, updated your comments and couldn't undo it. Dinakar is working on the strings.

File: extensions/resource-deployment/package.json [hunk 3]
```diff
@@ -13,16 +13,16 @@
     "azdata": "^1.6.0"
   },
   "activationEvents": [
-    "onCommand:azdata.resource.sql-image.deploy",
-    "onCommand:azdata.resource.sql-bdc.deploy"
+    "*"
```
Comment (alanrenmsft) [inline extensions/resource-deployment/package.json]: i had to change this to * due to https://github.com/microsoft/azuredatastudio/issues/5841

**Pull request merged**